JSON_UTIL.asp

```diff
@@ -1,15 +1,55 @@
 <%
-Function QueryToJSON(dbc, sql)
-        Dim rs, jsa
-        Set rs = dbc.Execute(sql)
-        Set jsa = jsArray()
-        While Not (rs.EOF Or rs.BOF)
-                Set jsa(Null) = jsObject()
-                For Each col In rs.Fields
-                        jsa(Null)(col.Name) = col.Value
-                Next
-        rs.MoveNext
-        Wend
-        Set QueryToJSON = jsa
-End Function
-%>
\ No newline at end of file
+Function QueryToJSON_(rs, propertyCase)
+  Dim rs, jsa, headers, headersBound, rows, rowsBound
+
+  Set jsa = jsArray()
+  if (NOT rs.EOF) AND (NOT rs.BOF) then
+  	headersBound = rs.fields.Count - 1
+  	redim headers(headersBound)
+
+    if (propertyCase = "upper") THEN
+      for i = 0 to headersBound
+        headers(i) = UCase(rs.fields(i).name)
+      Next
+    elseif (propertyCase = "lower") THEN
+      for i = 0 to headersBound
+        headers(i) = LCase(rs.fields(i).name)
+      Next
+    else
+      for i = 0 to headersBound
+        headers(i) = rs.fields(i).name
+      Next
+    end if
+
+    rows = rs.getrows()
+    rowsBound = UBound(rows, 2)
+
+    for rowIndex = 0 to rowsBound
+      set jsa(null) = jsObject()
+      for headerIndex = 0 to headersBound
+  			jsa(null)(headers(headerIndex)) = rows(headerIndex, rowIndex)
+  		next
+    next
+  end if
+
+  Set QueryToJSON_ = jsa
+End Function
+
+Function QueryToJSON(cmd)

```

Comment:
I think giving a different name (for example `PreparedQueryToJSON`) is better because this breaks previous implementation and some users can confuse who update from older version of code.